src/common/crc32c_ppc.c

```diff
@@ -35,8 +37,52 @@ static unsigned int crc32_align(unsigned int crc, unsigned char const *p,
 }
 #endif
 
-
 #ifdef HAVE_POWER8
+static inline unsigned long polynomial_multiply(unsigned int a, unsigned int b) {
+        vector unsigned int va = {a, 0, 0, 0};
+        vector unsigned int vb = {b, 0, 0, 0};
+        vector unsigned long vt;
+
+        __asm__("vpmsumw %0,%1,%2" : "=v"(vt) : "v"(va), "v"(vb));
+
+        return vt[0];
+}
+
+static unsigned int reflect32(unsigned int num) {
+        /*
+         * this can be improved, but for the simplicity of licensing I have
+         * written down the first correct implementation that came to mind
+         */
+        int i;
+        unsigned int result = 0;
+        for (i=0; i<32; i++) {
+                if (num & (1 << i))
```

Comment:
Since it is called twice per each ceph_crc32c_ppc, you may want to use more sophisticated implementation (one you have few lines below is already better).